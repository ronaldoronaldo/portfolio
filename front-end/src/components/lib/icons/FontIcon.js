import React from 'react'
import { Icon } from './FontIcon.styles'
import PropTypes from 'prop-types'

const FontIcon = ({ size, sizeMobile, color, iconName, ...rest }) => (
  <Icon
    size={size}
    sizeMobile={sizeMobile}
    color={color}
    className={`icon-${iconName}`}
    {...rest}
  />
)

export default FontIcon

FontIcon.propTypes = {
  iconName: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string
}
