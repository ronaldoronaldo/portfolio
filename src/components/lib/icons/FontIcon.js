import React from 'react'
import { Icon } from './FontIcon.styles'
import PropTypes from 'prop-types'

const FontIcon = ({ size, sizeMobile, hardMobile, color, iconName, ...rest }) => (
  <Icon
    hardMobile={hardMobile}
    size={size}
    sizeMobile={sizeMobile}
    color={color}
    className={`icon-${iconName}`}
    {...rest}
  />
)

export default FontIcon

FontIcon.propTypes = {
  iconName: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string
}
