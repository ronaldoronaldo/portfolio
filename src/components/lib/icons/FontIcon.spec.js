import React from 'react'
import { shallow } from 'enzyme'
import FontIcon from './FontIcon'

describe('<FontIcon />', () => {
  it('should be render correctly', () => {
    const wrapper = shallow(<FontIcon iconName='test' />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should be have a icon', () => {
    const wrapper = shallow(<FontIcon iconName='test' />)
    expect(wrapper.length).toBe(1)
  })
})