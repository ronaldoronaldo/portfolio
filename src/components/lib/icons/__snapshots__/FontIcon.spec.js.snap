// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FontIcon /> should be render correctly 1`] = `ShallowWrapper {}`;
